### PR TITLE
Add Emilia TTS recipe (ESPnet Bootcamp)

### PR DESCRIPTION
**Pull request by NewGamezzz**
## What did you change?

This pull request adds a new recipe for training text-to-speech (TTS) models on the Emilia dataset. The recipe is part of the WavLab bootcamp project.

---

## Why did you make this change?

The Emilia dataset, released in 2024, has recently gained attention and been used in several research papers. Adding support for this dataset to the ESPnet recipe will make it easier for researchers to experiment with and reproduce TTS results using Emilia.

---

## Is your PR small enough?

Yes

---

## Additional Context

Since the Emilia dataset does not include an official test set, the VCTK dataset is used as the evaluation set. The default model used in this recipe is the VITS model with x-vector conditioning.

Below are preliminary results from training the VITS model on the Emilia 100-hour subset for 1000 epochs (100,000 iterations):
  | utmos | wer % (Whisper) | speaker_similarity
-- | -- | -- | --
epoch 300 | 2.75 | 42.81% | 0.265
epoch 700 | 2.88 | 38.06% | 0.277
epoch 1000 | 3.03 | 32.01% | 0.287

Note that the speaker similarity may be lower than expected because the 100-hour subset contains a limited number of speakers, which may reduce diversity in speaker representation.

### CODE REVIEW

File: egs2/emilia/tts1/conf/versa.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  beam_size: 5
+  text_cleaner: whisper_basic
+
+# pesq related metrics
+# -- pesq: perceptual evaluation of speech quality
+- name: pesq
+
+# stoi related metrics
+# -- stoi: short-time objective intelligibility
+- name: stoi
```
Comment (ftshijt) [inline egs2/emilia/tts1/conf/versa.yaml]: For evaluation, I do not see pesq/stoi/singmos as valid metrics to use here. Please consider safely removing those.

File: egs2/emilia/tts1/local/data_download.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+echo "Downloading VCTK auxiliary labels..."
+if [ ! -e "${db_root}/VCTK-Corpus/lab" ]; then
+    cd "${db_root}" || exit 1;
+    git clone https://github.com/kan-bayashi/VCTKCorpusFullContextLabel.git
+    cp -r VCTKCorpusFullContextLabel/lab ./VCTK-Corpus
+    cd "${cwd}" || exit 1;
+    echo "Successfully downloaded label data."
+else
+    echo "Already exists. Skipped."
+fi
```
Comment (ftshijt) [inline egs2/emilia/tts1/local/data_download.sh:83]: I saw in the emilia paper that they have a emilia test set. May I ask why VCTK is also used here as test set?

File: egs2/emilia/tts1/run.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+train_config=conf/train.yaml
+inference_config=conf/decode.yaml
+
+cleaner=tacotron
+g2p=g2p_en
+
+./tts.sh \
+    --ngpu 4 \
+    --lang en \
```
Comment (ftshijt) [inline egs2/emilia/tts1/run.sh:28]: May I ask how you gonna change for other languages? It would be good to consider other options if possible

**Review by ftshijt — COMMENTED** [review]
Thanks! While we are exciting to see results, please also check the comments as follows.

File: egs2/emilia/tts1/run.sh (hunk 3 above)
Comment (NewGamezzz) [reply egs2/emilia/tts1/run.sh:28]: I have added a language argument. You can now specify the language using:
```bash
./run.sh --stage 1 --stop-stage 1 --local_data_opts "--lang ZH"
```
I have also updated the README with this instruction.

File: egs2/emilia/tts1/local/data_download.sh (hunk 2 above)
Comment (NewGamezzz) [reply egs2/emilia/tts1/local/data_download.sh:83]: It seems that the Emilia paper does not provide details on how the test set was split, and the official test set is also not available on [Hugging Face](https://huggingface.co/datasets/amphion/Emilia-Dataset). So, I used VCTK as the test set to be able to compare with others.

File: egs2/emilia/tts1/conf/versa.yaml (hunk 1 above)
Comment (NewGamezzz) [reply egs2/emilia/tts1/conf/versa.yaml]: Done. I have removed the metrics that are not related to TTS.

**Comment by sw005320** [comment]
Please fix https://github.com/espnet/espnet/actions/runs/19211946281/job/54915513367?pr=6291#step:8:178

File: egs2/emilia/tts1/run.sh (hunk 3 above)
Comment (ftshijt) [reply egs2/emilia/tts1/run.sh:28]: Great! Thanks for sharing the update

File: egs2/emilia/tts1/local/data_download.sh (hunk 2 above)
Comment (ftshijt) [reply egs2/emilia/tts1/local/data_download.sh:83]: Thanks! I would recommend contacting them if possible in that case.

**Comment by sw005320** [comment]
What is the current status of this PR?
There is no result.

**Comment by NewGamezzz** [comment]
I have completed full-scale training. The following results are obtained at epoch 375:

| Metric | Value |
|:--|--:|
| UTMOS | 2.49 |
| MCD | 17.47 |
| DNS (overall) | 3.09 |
| DNS (P.808) | 3.59 |
| Whisper WER | 36% |
| Whisper CER | 22% |
| SingMOS | 3.71 |
| Speaker similarity | 0.30 |

I have also listened to the generated utterances, and the model appears to be producing intelligible speech and attempting to generate meaningful content. I decided to stop training at this point as the model has begun to converge. I will add these results to the README for completeness.

**Comment by sw005320** [comment]
They look very bad to me.
Would more epochs mitigate the issue?

**Comment by NewGamezzz** [comment]
I will continue training the model for additional epochs to further assess whether the performance improves. If the results do not show an improvement, I will try to tune the hyperparameters and report the updated results as soon as possible.

**Comment by NewGamezzz** [comment]
I have completed the updates to the filtering process in the data preparation pipeline and the model configuration. The corresponding results have been updated in the README.

File: egs2/emilia/tts1/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# Changing the Language
+The Emilia dataset includes six languages: English (EN), Chinese (ZH), German (DE), French (FR), Japanese (JA), and Korean (KO). You can specify the language to train with:
+
+```bash
+./run.sh --stage 1 --stop-stage 1 --local_data_opts "--lang ZH"
+```
+
+The `--lang` argument must use one of the short codes listed above. When set, the dataset download and preprocessing steps will run for that language only. If not specified, the default language is English.
+
+# Results
```
Comment (sw005320) [inline egs2/emilia/tts1/README.md:34]: Can you also add a model link and environmental information?
e.g., https://github.com/espnet/espnet/blob/master/egs2/genshin/tts1/README.md#environments
(do not have to strictly follow the style, buu add the above information anyway).

**Comment by sw005320** [comment]
Thanks, @NewGamezzz!
